libyul/backends/evm/EVMDialect.cpp

```diff
@@ -140,18 +140,19 @@ map<YulString, BuiltinFunctionForEVM> createBuiltins(langutil::EVMVersion _evmVe
 				_assembly.appendAssemblySize();
 			else
 			{
+				auto [isData, subIdPath] = _context.currentObject->pathToSubObject(dataName.str());
 				yulAssert(
-					_context.subIDs.count(dataName) != 0,
+					!subIdPath.empty() | isData,
 					"Could not find assembly object <" + dataName.str() + ">."
 				);
-				_assembly.appendDataSize(_context.subIDs.at(dataName));
+				_assembly.appendDataSize(isData ? std::vector<size_t>{_context.subIDs[dataName]} : subIdPath);
```

Comment:
Am I right that `_context.subIDs[<dataName up to the first dot>] == subIdPath[0]` when `!isData` (which means that behavior did not change for non-nested objects)? If so, maybe an assert for that would be useful. It's not at all obvious because to know that you have to know where and how `_context.subIDs` is filled with data. I'm still not 100% sure I got that right.